src/utils/samlUtils.ts

```diff
@@ -77,6 +79,99 @@ const SAMLResponse = t.type({
   SAMLResponse: t.string
 });
 
+export const InfoNotAvailable = "NOT AVAILABLE";
+
+/**
+ * If an eventHandler and a feature flag are provided this function logs the timing deltas.
+ * This is useful to monitor the timings and to adjust the clockSkewMs variable
+ */
+export const extractAndLogTimings = (
+  startTime: number,
+  idpIssuer: string,
+  requestId: string,
+  eventHandler?: EventTracker,
+  hasDeltasLogging?: boolean
+) => (info: IIssueInstantWithAuthnContextCR): TE.TaskEither<never, void> => {
+  if (eventHandler && hasDeltasLogging) {
+    const extractNotOnOrAfterDelta = (
+      element: Element
+    ): E.Either<Error, string> =>
+      pipe(
+        NonEmptyString.decode(element.getAttribute("NotOnOrAfter")),
+        E.chain(UTCISODateFromString.decode),
+        E.mapLeft(() => new Error("Could not find/convert NotOnOrAfter")),
+        E.map(NotOnOrAfter => String(startTime - NotOnOrAfter.getTime()))
```

Comment:
`NotOnOrAfter` is greater than `startTime` or it should be.
```suggestion
        E.map(NotOnOrAfter => String(NotOnOrAfter.getTime() - startTime))
```

Reply:
We could just use Math.abs(NotOnOrAfter.getTime() - startTime).

Reply:
i too think it could be negative. the calculation wasn't correct before. solved in https://github.com/pagopa/io-spid-commons/pull/121/commits/abb3f88b8c3ddccf6e6f31e4b1d27d03b69dbbdb . thanks @BurnedMarshal

Reply:
I misunderstood @BurnedMarshal comment, I thought he was saying that it "could" and so to do both cases. In reality he was stating that it "must" be greater than start time. Fix is right.